src/System.Collections.Concurrent/src/System/Collections/Concurrent/PartitionerStatic.cs

```diff
@@ -246,8 +246,9 @@ public static OrderablePartitioner<TSource> Create<TSource>(IEnumerable<TSource>
             int coreOversubscriptionRate = 3;
 
             if (toExclusive <= fromInclusive) throw new ArgumentOutOfRangeException(nameof(toExclusive));
-            int rangeSize = (toExclusive - fromInclusive) /
-                (PlatformHelper.ProcessorCount * coreOversubscriptionRate);
+            long substraction = (long)toExclusive - fromInclusive;
+            int rangeSize = (int)(substraction /
+                (PlatformHelper.ProcessorCount * coreOversubscriptionRate));
```

Comment:
It'd be worth adding a note to the comment on coreOversubscriptionRate that it must be > 1.  Otherwise, if ProcessorCount == 1 and coreOverscriptionRate == 1, we'll be dividing by 1, and the same issue could occur with rangeSize == -1 when fromInclusive = int.MinValue and toExclusive = int.MaxValue.

Reply:
added, thank you.